src/Core/SettingsChangesHistory.cpp

```diff
@@ -83,6 +83,7 @@ static std::initializer_list<std::pair<ClickHouseVersion, SettingsChangesHistory
             {"input_format_binary_read_json_as_string", false, false, "Add new setting to read values of JSON type as JSON string in RowBinary input format"},
             {"min_free_disk_bytes_to_perform_insert", 0, 0, "New setting."},
             {"min_free_disk_ratio_to_perform_insert", 0.0, 0.0, "New setting."},
+            {"enable_named_columns_in_function_tuple", false, false, "Force disable the setting since it breaks queries"},
```

Comment:
This one is a bit confusing, shouldn't it be `true, false` since it was enabled in the previous version? But on the other hand, it's disabled in this new old version :/ So, what logic is applied here?

Reply:
> This one is a bit confusing, shouldn't it be true, false since it was enabled in the previous version?

That's what it was, but it's precisely what we want to change. The value shouldn't have been `true` due to the multiple bugs it introduced. I'm changing here to `false, false` so that even if you use `compatibility=24.8` it is disabled by default, since I intend to backport the change in default to previous versions too.

Note that technically this line shouldn't be necessary since I'm changing the old value in the 24.7 list, but I'm adding it here because otherwise the checks we have for setting changes will fail.